### ISSUE TITLE
Issues When Training with WebDatasets/Larger GPT2 Model

### ISSUE THREAD

**Issue by TheoCoombes** [rmokady/CLIP_prefix_caption#20]
I have attempted to train with the `gpt2-xl` model from huggingface as well as a custom dataloader that can preprocess webdataset archives, but am having issues when it comes to inference.

I have created a small script to test inference (`inference_test.py`) [on my fork](https://github.com/TheoCoombes/CLIP_prefix_caption) but it seems to repeatedly generate the same unrelated caption for any input (below are some samples from the RedCaps dataset):

![image](https://user-images.githubusercontent.com/17856206/150643748-175ccc61-d06a-46bf-aa5c-5b5e90212db3.png)

I'm not too sure what has gone wrong, but I believe it's either down to my dataloader not preprocessing the dataset in the correct way, or it's down to the model not supporting GPT2-xl.

* The preprocessing script is at `parse_webdatasets.py` and is modified from `rom1504/clip-retrieval`'s inference script. It does more of the transformations usually seen in train.py beforehand to optimise the training speed, and generates many `.npy` files as a result.
* These files are loaded with the custom dataloader on `train.py`.

Do you know what might be causing this?

Thanks in advance

**Comment by TheoCoombes** [comment]
I forgot to mention. The model has been trained using transformers rather than MLP.

These are the args used on my training script:
```
python3 train.py --only_prefix --data_dir "./train/" --out_dir "./model/" --prefix_length 10 --prefix_length_clip 10 --bs 15 --mapping_type "transformer" --device "cuda:7"
```

**Comment by rmokady** [comment]
Hi @TheoCoombes,

I'm not sure what is the issue, but another user introduce to me a similar outcome

In his case, the problem was the a bug in the preprocessing

I suggest the following:

1. Make sure training loss decreasing properly
2. Make sure the network don't output the same during training, print the result of the network on specific validation images
3. Check the exact validation images during inference

**Comment by rmokady** [comment]
I checked again the code, and with the MLP mapping and fine-tuning GPT-2 there caption are definitely different for each image.
